xplique/example_based/semifactuals.py

```diff
@@ -0,0 +1,233 @@
+        Distance function for examples search. It can be an integer, a string in
+        {"manhattan", "euclidean", "cosine", "chebyshev", "inf"}, or a Callable,
+        by default "euclidean".
+    """
+    # pylint: disable=duplicate-code
+
+    _returns_possibilities = [
+        "examples", "weights", "distances", "labels", "include_inputs",
+        "nuns", "nuns_indices", "dist_to_nuns", "nuns_labels"
+    ]
+
+    def __init__(
+        self,
+        cases_dataset: Union[tf.data.Dataset, tf.Tensor, np.ndarray],
+        targets_dataset: Union[tf.data.Dataset, tf.Tensor, np.ndarray],
+        labels_dataset: Optional[Union[tf.data.Dataset, tf.Tensor, np.ndarray]] = None,
+        k: int = 1,
+        projection: Union[Projection, Callable] = None,
+        case_returns: Union[List[str], str] = "examples",
+        batch_size: Optional[int] = 32,
+        distance: Union[int, str, Callable] = "euclidean",
+    ):
+
+        super().__init__(
+            cases_dataset=cases_dataset,
+            labels_dataset=labels_dataset,
+            targets_dataset=targets_dataset,
+            k=k,
+            projection=projection,
+            case_returns=case_returns,
+            batch_size=batch_size,
+        )
+
+        # set distance function and order for the search method
+        self.distance = distance
+        self.order = ORDER.ASCENDING
+
+        # initiate search_method
+        self.search_method = self.search_method_class(
+            cases_dataset=self.projected_cases_dataset,
+            targets_dataset=self.targets_dataset,
+            k=self.k,
+            search_returns=self._search_returns,
+            batch_size=self.batch_size,
+            distance=self.distance,
+        )
+
+    @property
+    def returns(self) -> Union[List[str], str]:
+        """Override the Base class returns' parameter."""
+        return self._returns
+
+    @returns.setter
+    def returns(self, returns: Union[List[str], str]):
+        """
+        Set the returns parameter. The returns parameter is a string
+        or a list of string with the elements to return in `self.explain()`.
+        Possibly returned elements are defined with `_returns_possibilities` static attribute.
+        """
+        default = "examples"
+        self._returns = _sanitize_returns(returns, self._returns_possibilities, default)
+        self._search_returns = ["indices", "distances"]
+
+        if isinstance(self._returns, list) and ("nuns" in self._returns):
+            self._search_returns.append("nuns_indices")
+        elif isinstance(self._returns, list) and ("nuns_indices" in self._returns):
+            self._search_returns.append("nuns_indices")
+        elif isinstance(self._returns, list) and ("nuns_labels" in self._returns):
+            self._search_returns.append("nuns_indices")
+
+        if isinstance(self._returns, list) and ("dist_to_nuns" in self._returns):
+            self._search_returns.append("dist_to_nuns")
```

Comment:
`self._returns` is always a list, the first condition can be removed. Or an assertion can be raised.